content/rs/databases/connect/endpoint-discovery.md

```diff
@@ -0,0 +1,125 @@
+
+{{<image filename="images/rs/endpoint-discovery/oss-cluster-api-discovery.png" width="75%" alt="The diagram shows how the client retrieves cluster information from a proxy and uses this information to connect to the other database endpoints.">}}{{</image>}}
+
+### Retrieve updated database topology
+
+If a clustered database's topology changes due to node failures or other updates, the client receives a `MOVED` response. The client can use the OSS Cluster API to retrieve an updated topology and then connect to the database through the new endpoints.
+
+{{<image filename="images/rs/endpoint-discovery/oss-cluster-api-failure.png" width="75%" alt="The diagram shows that the client connects to a different proxy to retrieve cluster information if a node fails.">}}{{</image>}}
+
+### OSS Cluster API considerations
+
+The OSS Cluster API has the following benefits:
+
+- If you migrate an open source Redis Cluster deployment to a Redis Enterprise Software database with the OSS Cluster API enabled, it functions the same as before migration.
+
+- Decreased cross-node network hops, which can improve performance in large throughput scenarios.
+
+{{<note>}}
+If you add more nodes to the cluster, this also increases the number of connections the client needs to open.
+{{</note>}}
+
+## Discovery service
+
+Redis Enterprise Software exposes a [Sentinel](https://redis.io/docs/management/sentinel/#sentinel-api)-compliant [discovery service]({{<relref "/rs/databases/durability-ha/discovery-service">}}) on port `8001`.
+
+Because the discovery service only returns one IP address, clients cannot easily discover all available endpoints when scaling a database with multiple nodes and proxies.
+
+See [Discovery service]({{<relref "/rs/databases/durability-ha/discovery-service">}}) for more information.
+
+### Use Sentinel commands
+
+Clients can use [Sentinel commands](https://redis.io/docs/management/sentinel/#sentinel-commands) to discover a database's proxy IP address and port.
+
+{{<image filename="images/rs/endpoint-discovery/sentinel-service-discovery.png" width="75%" alt="The diagram shows how a client uses the discovery service to identify the master IP address and uses it to connect to the proxy endpoint.">}}{{</image>}}
+
+### Node failures or changes
+
+If the database endpoint is bound to a different node due to failures or configuration changes, the discovery service returns a new IP address that clients can use to connect to the database.
+
+{{<image filename="images/rs/endpoint-discovery/sentinel-service-failure.png" width="75%" alt="The diagram shows how the discovery service sends the client a new master IP address after the original master node fails.">}}{{</image>}}
+
+## Load balancer
+
+If you did not set up DNS for your Redis Enterprise Software deployment, consider [setting up a load balancer]({{<relref "/rs/networking/cluster-lba-setup">}}) to direct traffic to the cluster nodes.
```

Comment:
Not technically wrong, but I might use "If you cannot set up DNS..."  to emphasize the preference.